packages/blade/src/components/Tooltip/TooltipContextWrapper.native.tsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+import type { CSSProperties } from 'react';
+import styled from 'styled-components/native';
+import type { EasingFn } from 'react-native-reanimated';
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withDelay,
+  withTiming,
+} from 'react-native-reanimated';
+import React from 'react';
+import type { View } from 'react-native';
+import { getTooltipContentWrapperStyles } from './getTooltipContentWrapperStyles';
+import type { TooltipContentWrapperProps } from './types';
+import BaseBox from '~components/Box/BaseBox';
+import { useTheme } from '~components/BladeProvider';
+import { size } from '~tokens/global';
+
+const StyledTooltipContentWrapper = styled(BaseBox)<{ collapse?: boolean; styles: CSSProperties }>(
+  ({ theme, styles }) => {
+    return getTooltipContentWrapperStyles({ theme, styles });
+  },
+);
+
+const TooltipContentWrapper = React.forwardRef<View, TooltipContentWrapperProps>(
+  ({ children, styles, side, isVisible, ...props }, ref) => {
+    const { theme } = useTheme();
+
+    const isOppositeAxis = side === 'right' || side === 'bottom';
+    const isHorizontal = side === 'left' || side === 'right';
+    const offset = isOppositeAxis ? -size[4] : size[4];
+
+    const translate = useSharedValue(offset);
+    const opacity = useSharedValue(0);
+
+    const easing = (theme.motion.easing.entrance.effective as unknown) as EasingFn;
+    const duration = theme.motion.duration.quick;
+
+    React.useEffect(() => {
```

Comment:
can you add some comments on why all these things are required?

Reply:
Do you mean 

"why do we need this useEffect" 
or 
"why do we need this `opacity.value = withDelay(duration, withTiming(isVisible ? 1 : 0, timings))`"